File: src/types/Treatment.ts
export interface TreatmentPhoto {
  id: string;
  treatmentId: string;
  photoUrl: string;
  uploadedDate: string;
}

export interface Treatment {
  id: string;
  customerId: string;
  treatmentTypeId: string;
  /** Denormalized from API response — treatment type name */
  treatmentTypeName?: string;
  treatmentSeriesId?: string;
  treatmentDate: string;
  durationMinutes?: number;
  /** Server-derived user id of the therapist who performed the treatment */
  userId?: string;
  /** Kept as alias for backward compatibility with mock data; use userId for API responses */
  therapistId?: string;
  /** Snapshot of the therapist's name at recording time */
  performedByFullName?: string;
  notes?: string;
  treatmentPhotos: TreatmentPhoto[];
}

File: backend/BeautyCareClinic.Domain/Entities/Treatment.cs
namespace BeautyCareClinic.Domain.Entities;

public class Treatment
{
    public Guid Id { get; set; }
    public Guid CustomerId { get; set; }
    public Guid TreatmentTypeId { get; set; }
    public Guid UserId { get; set; }
    public Guid? TreatmentSeriesId { get; set; }
    public DateTime TreatmentDate { get; set; }
    public int DurationMinutes { get; set; }
    public string? Notes { get; set; }
    public string PerformedByFullName { get; set; } = string.Empty;

    public Customer Customer { get; set; } = null!;
    public TreatmentType TreatmentType { get; set; } = null!;
    public User User { get; set; } = null!;
    public TreatmentSeries? TreatmentSeries { get; set; }
    public ICollection<TreatmentPhoto> Photos { get; set; } = new List<TreatmentPhoto>();
}
